backend/src/akvo/lumen/lib/aggregation/commons.clj

```diff
@@ -50,3 +51,40 @@
   (if (= "option" (:type bucket-column))
     (format "unnest(regexp_split_to_array(%1$s,'\\|'))" (:columnName bucket-column))
     (:columnName bucket-column)))
+
+(defn data-groups-sql-template
+  "Returns a data structure useful to generate a SELECT statement with the data groups passed as
+  parameter. We keep all columns for `metadata` groupId, and we *drop* `rnum` and `instance_id`
+  from all other data groups."
+  [data-groups]
+  (let [adapted-dgs (reduce
+                     (fn [dgs {:keys [columns groupId table-name]}]
+                       (conj dgs
+                             {:table-name table-name
+                              :columns (if (= groupId "metadata")
+                                         columns
+                                         (remove (fn [{:keys [columnName]}]
+                                                   (or (= "instance_id" columnName)
+                                                       (= "rnum" columnName)))
+                                                 columns))}))
+                     []
+                     data-groups)]
+    {:select (mapv :columnName (flatten (map :columns adapted-dgs)))
+     :from (mapv :table-name adapted-dgs)}))
+
+(defn data-groups-sql
+  [template]
```

Comment:
Is this really a template? Is it not the other way around that this function that kinda is the template and that what we pass in is some kind of "data"?